workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestQuery.cs

```diff
@@ -0,0 +1,117 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct ComponentInterest.Query objects.
+    /// </summary>
+    public class InterestQuery
+    {
+        private ComponentInterest.Query query;
+
+        // Hides the default constructor
+        private InterestQuery()
+        {
+        }
+
+        /// <summary>
+        ///     Creates an InterestQuery.
+        /// </summary>
+        /// <param name="constraint">
+        ///     A Constraint object defining the constraints of the query.
+        /// </param>
+        /// <remarks>
+        ///     Returns the full snapshot result by default.
+        /// </remarks>
+        /// <returns>
+        ///     An InterestQuery object.
+        /// </returns>
+        public static InterestQuery Query(Constraint constraint)
+        {
+            var interest = new InterestQuery
+            {
+                query =
+                {
+                    Constraint = constraint.AsQueryConstraint(),
+                    FullSnapshotResult = true,
+                    ResultComponentId = new List<uint>()
+                }
+            };
+            return interest;
+        }
+
+        /// <summary>
+        ///     Sets the maximum frequency of the query.
+        /// </summary>
+        /// <param name="frequency">
+        ///     The maximum frequency to return query results.
+        /// </param>
+        /// <returns>
+        ///     An updated InterestQuery object.
+        /// </returns>
+        public InterestQuery WithMaxFrequencyHz(float frequency)
+        {
+            query.Frequency = frequency;
+            return this;
+        }
+
+        /// <summary>
+        ///     Defines what components to return in the query results.
+        /// </summary>
+        /// <param name="resultComponentId">
+        ///     First ID of a component to return from the query results.
+        /// </param>
+        /// <param name="resultComponentIds">
+        ///     Further IDs of components to return from the query results.
+        /// </param>
+        /// <remarks>
+        ///     At least one component ID must be provided.
```

Comment:
Isn't this redundant, given the signature of the function?

Reply:
Perhaps yes, but I wanted to keep the remarks somewhat consistent between functions where this is implied by the signature and those that just take the `IEnumerable`